crates/pop-cli/src/commands/install/mod.rs

```diff
@@ -0,0 +1,338 @@
+	#[clap(short('y'), long)]
+	skip_confirm: bool,
+}
+
+impl InstallArgs {
+	pub(crate) async fn execute(self) -> anyhow::Result<()> {
+		clear_screen()?;
+		set_theme(Theme);
+		intro(format!(
+			"{}: Install dependencies for development",
+			style(" Pop CLI ").black().on_magenta()
+		))?;
+		if cfg!(target_os = "macos") {
+			log::info("ℹ️ Mac OS (Darwin) detected.")?;
+			install_mac(self.skip_confirm).await?;
+		} else if cfg!(target_os = "linux") {
+			match os_info::get().os_type() {
+				Type::Arch => {
+					log::info("ℹ️ Arch Linux detected.")?;
+					install_arch(self.skip_confirm).await?;
+				},
+				Type::Debian => {
+					log::info("ℹ️ Debian Linux detected.")?;
+					install_debian(self.skip_confirm).await?;
+				},
+				Type::Redhat => {
+					log::info("ℹ️ Redhat Linux detected.")?;
+					install_redhat(self.skip_confirm).await?;
+				},
+				Type::Ubuntu => {
+					log::info("ℹ️ Ubuntu detected.")?;
+					install_ubuntu(self.skip_confirm).await?;
+				},
+				_ => return not_supported_message(),
+			}
+		} else {
+			return not_supported_message();
+		}
+		install_rustup().await?;
+		outro("✅ Installation complete.")?;
+		Ok(())
+	}
+}
+
+async fn install_mac(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/macos/")?;
+	if !skip_confirm {
+		prompt_for_confirmation(&format!(
+			"{}, {}, {}, {} and {}",
+			Dependencies::Homebrew,
+			Dependencies::Protobuf,
+			Dependencies::Openssl,
+			Dependencies::Rustup,
+			Dependencies::Cmake,
+		))?
+	}
+	install_homebrew().await?;
+	cmd("brew", vec!["update"]).run()?;
+	cmd(
+		"brew",
+		vec![
+			"install",
+			&Dependencies::Protobuf.to_string(),
+			&Dependencies::Openssl.to_string(),
+			&Dependencies::Cmake.to_string(),
+		],
+	)
+	.run()?;
+
+	Ok(())
+}
+
+async fn install_arch(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/linux/")?;
+	if !skip_confirm {
+		prompt_for_confirmation(&format!(
+			"{}, {}, {}, {}, {} and {}",
+			Dependencies::Curl,
+			Dependencies::Git,
+			Dependencies::Clang,
+			Dependencies::Make,
+			Dependencies::Openssl,
+			Dependencies::Rustup,
+		))?
+	}
+	cmd("pacman", vec!["-Syu", "--needed", "--noconfirm", &Dependencies::Openssl.to_string()])
+		.run()?;
+	cmd(
+		"pacman",
+		vec![
+			"-Syu",
+			"--needed",
+			"--noconfirm",
+			&Dependencies::Curl.to_string(),
+			&Dependencies::Git.to_string(),
+			&Dependencies::Clang.to_string(),
+			&Dependencies::Make.to_string(),
+		],
+	)
+	.run()?;
+
+	Ok(())
+}
+async fn install_ubuntu(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/linux/")?;
+	if !skip_confirm {
+		prompt_for_confirmation(&format!(
+			"{}, {}, {}, {}, {} and {}",
+			Dependencies::Git,
```

Comment:
Fine for now, but in future you could define all the required items in a const array at the top of the function and then just use them as a set when required.

Reply:
I considered it, but the legibility of the code seemed worse to me, for example:
```
const MACOS_PACKAGES: [Dependencies; 5] = [Homebrew, Protobuf, Openssl, Rustup, Cmake];
cmd("brew",vec!["install",&MACOS_PACKAGES[1].to_string(),&MACOS_PACKAGES[2].to_string(),&MACOS_PACKAGES[4].to_string(),],).run()?;
```

Reply:
I meant something simpler like (not tested): 
`cmd("brew",vec!["install"].concat(PACKAGES.iter().map(|d| d.to_string()).run()?;`

You also dont need to prefix it with `MACOS_` if the const is defined within the function.

Reply:
Doesn't look bad, However we will need to mantain a few constants. For example, for MAC:
```
const PACKAGES: [Dependencies; 5] = [Homebrew, Protobuf, Openssl, Rustup, Cmake];
prompt_for_confirmation(&format!("{}", PACKAGES.join(""))
```
but to install with `brew` only need 3 of them, so we will need: 
```
const PACKAGES_BREW_INSTALL: [Dependencies; 3] = [Protobuf, OpenssL,, Cmake];
cmd("brew",vec!["install"].concat(PACKAGES.iter().map(|d| d.to_string()).run()?;
```

Reply:
Constants dont have to be defined at the top of the module, they can be defined within the relevant function. It is simply so that you define the list of packages once, then for the confirmation and the install you use that same set of packages. Then if you need to add/remove a package, you just add it to the const array and job done.

Reply:
Just .iter().skip(2)

Reply:
As mentioned, fine to ignore for now. Its just easiest to define things once and use them in many places to save having to update stuff in multiple places every time there might be a change.

Reply:
Thanks for the feedback. I was just trying to fully understand it. I'll leave things as they are for now, but I'll consider this refactor with proper testing.